src/main/java/duke/command/Command.java

```diff
@@ -21,12 +21,12 @@ public boolean isExit() {
     }
 
     /**
-     * Executes the command.
-     *
+     * Executes the command and returns and acknowledgement message.
      * @param tasks The TaskList containing the tasks.
      * @param ui The Ui that interacts with the user.
      * @param storage The Storage to load and save tasks into the data file.
+     * @return A string with the message to be printed.
      * @throws DukeException If the command is invalid.
      */
-    public abstract void execute(TaskList tasks, Ui ui, Storage storage) throws DukeException;
+    public abstract String execute(TaskList tasks, Ui ui, Storage storage) throws DukeException;
```

Comment:
Good use of abstract class. Implementation of Command class promotes better OOP.
Classes that implemented the execute function have a clear instruction on what is being done.